backend/src/main/kotlin/me/andrewda/utils/Models.kt

```diff
@@ -0,0 +1,109 @@
+package me.andrewda.utils
+
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.jetbrains.exposed.dao.Entity
+import org.jetbrains.exposed.dao.EntityID
+import kotlin.reflect.full.memberProperties
+
+enum class ReadLevel {
+    USER, SELF, ADMIN
+}
+
+/**
+ * Marks a field as readable in the API response.
+ *
+ * @param key the key to use in the response (defaults to the variable name)
+ * @param readLevel the [ReadLevel] required to read this value
+ * @param deep whether this field is a deep (performs a database query)
+ */
+annotation class Readable(
+    val key: String = "",
+    val readLevel: ReadLevel = ReadLevel.USER,
+    val deep: Boolean = false
+)
+
+/**
+ * Gets the [ApiResponse] generated for the [Entity] given a specific [readLevel].
+ *
+ * @param readLevel the permission level at which the response should be generated
+ * @return the generated [ApiResponse]
+ */
+fun Entity<*>.getApiResponse(readLevel: ReadLevel = ReadLevel.USER): ApiResponse {
+    val result = mutableMapOf<String, Any?>()
```

Comment:
It's not immediately obvious that `mutableMapOf<String, Any?>()` makes an `ApiResponse`, so it might be better to make a function in utils called `makeResponse()` or something like that.

Reply:
`ApiResponse` is just a typealias for `MutableMap<String, Any?>`. Would your `makeResponse` method just cast the latter to the former?